packages/itwinui-react/src/core/Table/Table.tsx

```diff
@@ -597,6 +596,19 @@ export const Table = <
     visibleColumns,
     setGlobalFilter,
   } = instance;
+
+  let headerGroups = _headerGroups;
+
+  if (columns.length === 1 && 'columns' in columns[0]) {
+    headerGroups = _headerGroups.slice(1);
+    if (isDev && !didLogWarning) {
+      console.warn(
+        `Table's \`columns\` prop should not have a top-level Header or sub-columns. It is only kept for backwards compatibility.\n See https://github.com/iTwin/iTwinUI/wiki/iTwinUI-react-v2-migration-guide#breaking-changes`,
```

Comment:
Message should be the same as in config jsdocs.

Reply:
updated a little bit. is it better now?

for users who don't know how it was before major release, they will use the new columns type anyway, so they won't see the message